### PR TITLE
Todo Crud

### PR DESCRIPTION
**Pull request by MosDevx**
Changes Implemented:
    Remove all hardcoded items from the tasks array.
    Create a new JavaScript file for the new functionality.
    Implement a function for adding a new task (add a new element to the array).
    Implement a function for deleting a task (remove an element from the array).
    Implement a function for editing task descriptions.
    By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
    Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
    All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/Todo.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+// const todo1 = new Todo('abc');
+// const todo2 = new Todo('def');
+// const todo3 = new Todo('ijk');
+
+// console.log(Todo.getAllTodos());
+// todo2.deleteTodo()
+// const todo4 = new Todo('lmn');
+// todo4.updateTodo("Just did it")
+// console.log(Todo.getAllTodos());
```
Comment (Wes-Isaac) [inline src/Todo.js]: - [ ] Great job so far, kindly remove commented code to keep the code base clean.

File: src/Todo.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if (typeof Todo.storageUpdater === 'function') {
+      Todo.storageUpdater();
+    }
+  }
+
+  updateTodo(newTask) {
+    const i = Todo.todoArray.indexOf(this);
+    const todo = Todo.todoArray[i];
+    todo.task = newTask;
+  }
```
Comment (Wes-Isaac) [inline src/Todo.js]: - [ ] Great job implementing all the necessary functions, however when after I edit it doesn't store the data on local storage, kindly fix that.

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @MosDevx, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Great PR description 👍.
- Professional README 💯.
- All linters are passing 🥇.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by rustam-tolipov — APPROVED** [review]
Hi @MosDevx , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**